### DISCUSSION TITLE
Can a `LanguageDetector` be called from several threads in parallel ?

### DISCUSSION THREAD

**Discussion by rgeronimi** [pemistahl/lingua-py#244] [Q&A] [answered]
`lingua-py`'s documentation doesn't state if a `LanguageDetector` object can be called in parallel from several python threads (for distinct pieces of text of course).

In Python's ecosystem this is far from a guaranteed behavior unfortunately.

Is it the case ? Or should we use a `threading.Lock` to queue the calls to the same `LanguageDetector`?

**Comment by pemistahl** [comment, marked as answer]
Yes, both Rust and pure Python implementations are thread-safe. They use thread-safe data structures internally, so you can use a single `LanguageDetector` instance from multiple threads.

If you encounter any problems, then please open an issue with code to reproduce.

**Comment by rgeronimi** [comment]
Thanks this is great!
